### PR TITLE
Disable DJANGO_ALLOW_ASYNC_UNSAFE for tasking system

### PR DESCRIPTION
**Pull request by dralley**
This should no longer be necessary after all the work done to reduce db connections - and if we do encounter issues, we'd be better off trying to address them.

[noissue]

### CODE REVIEW

**Comment by dralley** [comment]
I don't think we can turn it off for the content app yet, we'll need to do the same work there that was done for the tasking system (for plugins like ansible, rpm, container)

File: pulpcore/plugin/stages/artifact_stages.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ async def run(self):
             # swap it out with the existing one.
             for digest_type, digests in artifact_digests_by_type.items():
                 query_params = {"{attr}__in".format(attr=digest_type): digests}
-                existing_artifacts_qs = Artifact.objects.filter(**query_params).only(digest_type)
+                existing_artifacts_qs = Artifact.objects.filter(**query_params)
```
Comment (dralley) [inline pulpcore/plugin/stages/artifact_stages.py:90]: Other values are used down on line 391, triggering DB queries.

**Comment by bmbouter** [comment]
Should we have a refactor issue to track this change? I'm not sure, but I wanted to ask.

**Comment by dralley** [comment]
I think yes - if we *potentially* might want to backport it to 3.15, otherwise no strong feeling either way.

**Comment by bmbouter** [comment]
> I think yes - if we _potentially_ might want to backport it to 3.15, otherwise no strong feeling either way.

+1 then, we may want to backport.

**Comment by dralley** [comment]
@bmbouter Done

**Review by bmbouter — APPROVED** [review]
This looks good to me!

What's the plan for the other areas that `DJANGO_ALLOW_ASYNC_UNSAFE` is used?

**Comment by dralley** [comment]
I filed https://pulp.plan.io/issues/9354 as well.

**Review by dkliban — APPROVED** [review]

**Comment by fao89** [comment]
pulp-ansible and pulp-container are failing now,
https://github.com/pulp/pulp_ansible/runs/3555934111?check_suite_focus=true
https://github.com/pulp/pulp_container/actions/runs/1215755815
I'm not sure yet if the plugins need a change, or if it is pulpcore that needs it

**Comment by dralley** [comment]
@fao89 The plugins - this means they didn't have complete mitigations for the DB connection issue FYI.  But it should be quite easy to find where the problems are since it throws exceptions.

**Pull request merged**